runtime/ops/command.rs

```diff
@@ -0,0 +1,359 @@
+
+  let mut child = command.spawn()?;
+  let pid = child.id().expect("Process ID should be set.");
+
+  let stdin_rid = child
+    .stdin
+    .take()
+    .map(|stdin| state.resource_table.add(ChildStdinResource::from(stdin)));
+
+  let stdout_rid = child
+    .stdout
+    .take()
+    .map(|stdout| state.resource_table.add(ChildStdoutResource::from(stdout)));
+
+  let stderr_rid = child
+    .stderr
+    .take()
+    .map(|stderr| state.resource_table.add(ChildStderrResource::from(stderr)));
+
+  let child_rid = state
+    .resource_table
+    .add(ChildResource(AsyncRefCell::new(child)));
+
+  Ok(Child {
+    rid: child_rid,
+    pid,
+    stdin_rid,
+    stdout_rid,
+    stderr_rid,
+  })
+}
+
+fn op_command_child_status(
+  state: &mut OpState,
+  rid: ResourceId,
+  _: (),
+) -> Result<Option<CommandStatus>, AnyError> {
+  let resource = state.resource_table.get::<ChildResource>(rid)?;
+  let mut child = RcRef::map(resource, |r| &r.0).try_borrow_mut().unwrap();
+  Ok(child.try_wait()?.map(|status| status.into()))
+}
+
+async fn op_command_child_wait(
+  state: Rc<RefCell<OpState>>,
+  rid: ResourceId,
+  stdin_rid: Option<ResourceId>,
+) -> Result<CommandStatus, AnyError> {
+  let resource = state
+    .borrow_mut()
+    .resource_table
+    .take::<ChildResource>(rid)?;
+  let mut child = RcRef::map(resource, |r| &r.0).borrow_mut().await;
+  if let Some(stdin_rid) = stdin_rid {
+    let stdin = state
+      .borrow_mut()
+      .resource_table
+      .take::<ChildStdinResource>(stdin_rid)?;
+    child.stdin = Some(Rc::try_unwrap(stdin).unwrap().into_inner());
+  }
+  Ok(child.wait().await?.into())
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct ChildStdio {
+  rid: ResourceId,
+  stdout_rid: Option<ResourceId>,
+  stderr_rid: Option<ResourceId>,
+}
+
+async fn op_command_child_output(
+  state: Rc<RefCell<OpState>>,
+  args: ChildStdio,
+  _: (),
+) -> Result<CommandOutput, AnyError> {
+  let resource = state
+    .borrow_mut()
+    .resource_table
+    .take::<ChildResource>(args.rid)?;
+  let resource = Rc::try_unwrap(resource).ok().unwrap();
+  let mut child = resource.0.into_inner();
+
+  if let Some(stdout_rid) = args.stdout_rid {
+    let stdout = state
+      .borrow_mut()
+      .resource_table
+      .take::<ChildStdoutResource>(stdout_rid)?;
+    child.stdout = Some(Rc::try_unwrap(stdout).unwrap().into_inner());
+  }
+  if let Some(stderr_rid) = args.stderr_rid {
+    let stderr = state
+      .borrow_mut()
+      .resource_table
+      .take::<ChildStderrResource>(stderr_rid)?;
+    child.stderr = Some(Rc::try_unwrap(stderr).unwrap().into_inner());
+  }
+
+  let output = child.wait_with_output().await?;
+
+  Ok(CommandOutput {
+    status: output.status.into(),
+    stdout: if args.stdout_rid.is_some() {
+      Some(output.stdout.into())
+    } else {
+      None
+    },
+    stderr: if args.stderr_rid.is_some() {
+      Some(output.stderr.into())
+    } else {
+      None
+    },
```

Comment:
It's not wrong but you can write this more succinctly as follows:
```suggestion
    stdout: args.stdout_rid.then(|| output.stdout.into()),
    stderr: args.stderr_rid.then(|| output.stderr.into()),
```